File: frontend/src/app/core/services/rendezvous.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface RendezVous {
  id: number;
  animalId: number;
  animalName: string;
  animalRace: string;
  animalGender: string;
  animalVaccinated: boolean;
  animalHealthStatus: string;
  dateTime: string;
  status: string;
}

@Injectable({
  providedIn: 'root'
})
export class RendezVousService {
  private API_URL = 'http://localhost:8088/api/rendezvous';

  constructor(private http: HttpClient) {}

  getRendezVousByUser(userId: number): Observable<any[]> {
    return this.http.get<any[]>(`${this.API_URL}/user/${userId}`);
  }
  getAllRendezVous(): Observable<RendezVous[]> {
    return this.http.get<RendezVous[]>(this.API_URL);
  }

  createRendezVous(rendezVous: RendezVous): Observable<RendezVous> {
    return this.http.post<RendezVous>(this.API_URL, rendezVous);
  }

  acceptRendezVous(rendezVousId: number): Observable<void> {
    return this.http.put<void>(`${this.API_URL}/${rendezVousId}/accept`, {});
  }
  cancelRendezVous(rendezVousId: number): Observable<void> {
    return this.http.put<void>(`${this.API_URL}/${rendezVousId}/cancel`, {});
  }

}

File: backend/src/main/java/com/animo/animorise/service/RendezVousService.java
package com.animo.animorise.service;

import com.animo.animorise.dto.RendezVousDto;

import java.time.LocalDateTime;
import java.util.List;

public interface RendezVousService {
    RendezVousDto createRendezVous(Long animalId, LocalDateTime dateTime, Long userId);
    RendezVousDto acceptRendezVous(Long rendezVousId);
    RendezVousDto cancelRendezVous(Long rendezVousId);
    List<RendezVousDto> getRendezVousByAnimal(Long animalId);

    List<RendezVousDto> getRendezVousByUser(Long userId);

    List<RendezVousDto> getAllRendezVous();
}